desktop/core/src/desktop/lib/importer/serializers_tests.py

```diff
@@ -0,0 +1,373 @@
+
+    assert not serializer.is_valid()
+    assert "import_type" in serializer.errors
+
+  def test_invalid_import_type(self):
+    invalid_data = {
+      "file_path": "/path/to/file.csv",
+      "import_type": "invalid_type",  # Not one of 'local' or 'remote'
+    }
+
+    serializer = GuessFileMetadataSerializer(data=invalid_data)
+
+    assert not serializer.is_valid()
+    assert "import_type" in serializer.errors
+    assert serializer.errors["import_type"][0] == '"invalid_type" is not a valid choice.'
+
+
+class TestPreviewFileSerializer:
+  def test_valid_csv_data(self):
+    valid_data = {
+      "file_path": "/path/to/file.csv",
+      "file_type": "csv",
+      "import_type": "local",
+      "sql_dialect": "hive",
+      "has_header": True,
+      "field_separator": ",",
+    }
+
+    serializer = PreviewFileSerializer(data=valid_data)
+
+    assert serializer.is_valid(), f"Serializer validation failed: {serializer.errors}"
+    # Check that default values are set for quote_char and record_separator
+    assert serializer.validated_data["quote_char"] == '"'
+    assert serializer.validated_data["record_separator"] == "\n"
+
+  def test_valid_excel_data(self):
+    valid_data = {
+      "file_path": "/path/to/file.xlsx",
+      "file_type": "excel",
+      "import_type": "local",
+      "sql_dialect": "hive",
+      "has_header": True,
+      "sheet_name": "Sheet1",
+    }
+
+    serializer = PreviewFileSerializer(data=valid_data)
+
+    assert serializer.is_valid(), f"Serializer validation failed: {serializer.errors}"
+
+  def test_missing_required_fields(self):
+    # Test with minimal data
+    invalid_data = {
+      "file_path": "/path/to/file.csv",
+    }
+
+    serializer = PreviewFileSerializer(data=invalid_data)
+
+    assert not serializer.is_valid()
+    # Check that all required fields are reported as missing
+    for field in ["file_type", "import_type", "sql_dialect", "has_header"]:
+      assert field in serializer.errors
+
+  def test_invalid_file_type(self):
+    invalid_data = {
+      "file_path": "/path/to/file.json",
+      "file_type": "json",  # Not a valid choice
+      "import_type": "local",
+      "sql_dialect": "hive",
+      "has_header": True,
+    }
+
+    serializer = PreviewFileSerializer(data=invalid_data)
+
+    assert not serializer.is_valid()
+    assert "file_type" in serializer.errors
+    assert serializer.errors["file_type"][0] == '"json" is not a valid choice.'
+
+  def test_excel_without_sheet_name(self):
+    invalid_data = {
+      "file_path": "/path/to/file.xlsx",
+      "file_type": "excel",
+      "import_type": "local",
+      "sql_dialect": "hive",
+      "has_header": True,
+      # Missing sheet_name
+    }
+
+    serializer = PreviewFileSerializer(data=invalid_data)
+
+    assert not serializer.is_valid()
+    assert "sheet_name" in serializer.errors
+    assert serializer.errors["sheet_name"][0] == "Sheet name is required for Excel files."
```

Comment:
Spreadsheet or Excel file? What happen if there multiple sheets?

Reply:
There can be multiple sheets in an Excel file. In the importer flow, users can create table from one particular sheet.